aws_advanced_python_wrapper/host_monitoring_v2_plugin.py

```diff
@@ -0,0 +1,506 @@
+
+    def is_active(self) -> bool:
+        connection_weak_ref = self._connection_to_abort.get()
+        return connection_weak_ref is not None and connection_weak_ref() is not None
+
+
+class HostMonitorV2:
+    """
+        This class uses a background thread to monitor a particular server with one or more active :py:class:`Connection`
+        objects. It performs periodic health checks and aborts connections when the server becomes unhealthy.
+        """
+    _THREAD_SLEEP_NANO = 100_000_000
+    _MONITORING_PROPERTY_PREFIX = "monitoring-"
+    _QUERY = "SELECT 1"
+
+    def __init__(
+            self,
+            plugin_service: PluginService,
+            host_info: HostInfo,
+            props: Properties,
+            failure_detection_time_ms: int,
+            failure_detection_interval_ms: int,
+            failure_detection_count: int,
+            aborted_connection_counter: TelemetryCounter):
+        self._plugin_service: PluginService = plugin_service
+        self._host_info: HostInfo = host_info
+        self._props: Properties = props
+        self._telemetry_factory: TelemetryFactory = self._plugin_service.get_telemetry_factory()
+        self._failure_detection_time_ns: int = failure_detection_time_ms * 10**6
+        self._failure_detection_interval_ns: int = failure_detection_interval_ms * 10**6
+        self._failure_detection_count: int = failure_detection_count
+        self._aborted_connection_counter: TelemetryCounter = aborted_connection_counter
+
+        self._active_contexts: Queue = Queue()
+        self._new_contexts: ConcurrentDict[float, Queue] = ConcurrentDict()
+        self._is_stopped: AtomicBoolean = AtomicBoolean(False)
+        self._is_unhealthy: bool = False
+        self._failure_count: int = 0
+        self._invalid_node_start_time_ns: int = 0
+        self._monitoring_connection: Optional[Connection] = None
+        self._driver_dialect: DriverDialect = self._plugin_service.driver_dialect
+
+        self._monitor_run_thread: Thread = Thread(daemon=True, name="HostMonitoringThreadRun", target=self.run)
+        self._monitor_run_thread.start()
+        self._monitor_new_context_thread: Thread = Thread(daemon=True, name="HostMonitoringThreadNewContextRun",
+                                                          target=self._new_context_run)
+        self._monitor_new_context_thread.start()
+
+    def can_dispose(self) -> bool:
+        return self._active_contexts.empty() and len(self._new_contexts.items()) == 0
+
+    @property
+    def is_stopped(self):
+        return self._is_stopped.get()
+
+    def stop(self):
+        self._is_stopped.set(True)
+
+    def start_monitoring(self, context: MonitoringContext):
+        if self.is_stopped:
+            logger.warning("HostMonitorV2.MonitorIsStopped", self._host_info.host)
+
+        current_time_ns = perf_counter_ns()
+        start_monitoring_time_ns = ((current_time_ns + self._failure_detection_time_ns) // 10**9) * 10**9
```

Comment:
```suggestion
        start_monitoring_time_ns = current_time_ns + self._failure_detection_time_ns
```
Can we create helper methods for these time conversions